src/handlers/push/update-base.ts

```diff
@@ -0,0 +1,36 @@
+import { Context } from "probot";
+import { getLogger } from "../../bindings";
+import { getPreviousFileContent, listLabelsForRepo, updateLabelsFromBaseRate } from "../../helpers";
+import { Label, PushPayload } from "../../types";
+import { parseYAML } from "../../utils/private";
+
+export const updateBaseRate = async (context: Context, payload: PushPayload, filePath: string) => {
+  const logger = getLogger();
+  // Get default branch from ref
+  const branch = payload.ref?.split("refs/heads/")[1];
+  const owner = payload.repository.owner.login;
+  const repo = payload.repository.name;
+
+  // get previous config
+  const preFileContent = await getPreviousFileContent(owner, repo, branch, filePath);
+
+  const previousContent = Buffer.from(preFileContent!, "base64").toString();
+  const previousConfig = await parseYAML(previousContent);
+
+  if (!previousConfig["base-multiplier"]) {
+    logger.debug("No multiplier found in file object");
+    return;
+  }
+
+  const previousBaseRate = previousConfig["base-multiplier"];
+
+  // fetch all labels
+  const repoLabels = await listLabelsForRepo();
+
+  if (repoLabels.length === 0) {
+    logger.debug("No labels on this repo");
+    return;
+  }
+
+  updateLabelsFromBaseRate(owner, repo, context, repoLabels as Label[], previousBaseRate);
```

Comment:
```suggestion
 await updateLabelsFromBaseRate(owner, repo, context, repoLabels as Label[], previousBaseRate);
```

I noticed that this is an async function. Shouldn't this be `await`ed?

Reply:
Yea, it can but this is the last function so it doesn't matter if there's an await or not

Reply:
The runtime can terminate before the response arrives unless you `await`.

Reply:
oh good catch @pavlovcik. it needs to be `await`

Reply:
> oh good catch @pavlovcik. it needs to be `await`

Awaited it

Reply:
> oh good catch @pavlovcik. it needs to be `await`. Even if it is the last function, it should be `await` because the bot runs as a lambda function so needs to wait until all the functions get through. If not, it just terminates without running. lets accept the request changes @devpanther 

Yea @0xcodercrane, I've Awaited it..

Reply:
> please post the commit link and mark it resolved

https://github.com/ubiquity/ubiquibot/pull/352/commits/de72859b8e7c27a82f97c6271dec55c8654b9142